h/views/client.py

```diff
@@ -49,6 +49,9 @@ def sidebar_app(request, extra=None):
     app_config = {
         'apiUrl': request.route_url('api.index'),
         'authDomain': request.authority,
+
+        # OAuth config.
+        'oauthAuthorizeUrl': request.route_url('oauth_authorize'),
```

Comment:
Ideally, yes.

However, there is currently an awkward circular dependency in the client that this URL is needed before authorization, but fetching `/api/links` in the client depends on first completing authorization. The solution I think will be to rework the client so that it can explicitly make certain requests without authenticating.